temporalcli/commands.schedule_test.go

```diff
@@ -0,0 +1,361 @@
+			Stdout: io.Discard,
+			Stderr: io.Discard,
+			Args: []string{
+				"schedule", "delete",
+				"--address", s.Address(),
+				"-s", s.schedId(),
+			},
+			Fail: func(error) {},
+		}
+		temporalcli.Execute(ctx, options)
+	})
+	return s.Execute(append([]string{
+		"schedule", "create",
+		"--address", s.Address(),
+		"-s", s.schedId(),
+		"--task-queue", s.Worker.Options.TaskQueue,
+		"--type", "DevWorkflow",
+		"--workflow-id", s.schedWfId(),
+	}, args...)...,
+	)
+}
+
+func (s *SharedServerSuite) TestSchedule_Create() {
+	res := s.createSchedule("--interval", "10d")
+	s.NoError(res.Err)
+}
+
+func (s *SharedServerSuite) TestSchedule_Delete() {
+	res := s.createSchedule("--interval", "10d")
+	s.NoError(res.Err)
+
+	// check exists
+	res = s.Execute(
+		"schedule", "describe",
+		"--address", s.Address(),
+		"-s", s.schedId(),
+	)
+	s.NoError(res.Err)
+
+	res = s.Execute(
+		"schedule", "delete",
+		"--address", s.Address(),
+		"-s", s.schedId(),
+	)
+	s.NoError(res.Err)
+
+	// doesn't exist anymore
+	res = s.Execute(
+		"schedule", "describe",
+		"--address", s.Address(),
+		"-s", s.schedId(),
+	)
+	s.Error(res.Err)
+}
+
+func (s *SharedServerSuite) TestSchedule_Describe() {
+	res := s.createSchedule("--interval", "2s")
+	s.NoError(res.Err)
+
+	time.Sleep(4 * time.Second) // run at least once
+
+	// text
+
+	res = s.Execute(
+		"schedule", "describe",
+		"--address", s.Address(),
+		"-s", s.schedId(),
+	)
+	s.NoError(res.Err)
+	out := res.Stdout.String()
+	s.ContainsOnSameLine(out, "ScheduleId", s.schedId())
+	s.ContainsOnSameLine(out, "Spec", "2s")
+	s.ContainsOnSameLine(out, "RunningWorkflows", s.schedWfId()+"-")
+
+	// json
+
+	res = s.Execute(
+		"schedule", "describe",
+		"--address", s.Address(),
+		"-s", s.schedId(),
+		"-o", "json",
+	)
+	s.NoError(res.Err)
+	var j struct {
+		Schedule struct {
+			Action struct {
+				StartWorkflow struct {
+					Id string `json:"workflowId"`
+				} `json:"startWorkflow"`
+			} `json:"action"`
+		} `json:"schedule"`
+	}
+	s.NoError(json.Unmarshal(res.Stdout.Bytes(), &j))
+	s.Equal(s.schedWfId(), j.Schedule.Action.StartWorkflow.Id)
+}
+
+func (s *SharedServerSuite) TestSchedule_CreateDescribeCalendar() {
+	res := s.createSchedule("--calendar", `{"hour":"2,4","dayOfWeek":"thu,fri"}`)
+	s.NoError(res.Err)
+
+	res = s.Execute(
+		"schedule", "describe",
+		"--address", s.Address(),
+		"-s", s.schedId(),
+	)
+	s.NoError(res.Err)
+	out := res.Stdout.String()
+	s.ContainsOnSameLine(out, "ScheduleId", s.schedId())
+	s.ContainsOnSameLine(out, "Spec", "dayOfWeek")
+}
+
+func (s *SharedServerSuite) TestSchedule_List() {
+	res := s.createSchedule("--interval", "10d")
+	s.NoError(res.Err)
+
+	// table
+
+	s.Eventually(func() bool {
+		res = s.Execute(
+			"schedule", "list",
+			"--address", s.Address(),
+		)
+		s.NoError(res.Err)
+		out := res.Stdout.String()
+		return AssertContainsOnSameLine(out, s.schedId(), "DevWorkflow", "false") == nil
+	}, 10*time.Second, time.Second)
```

Comment:
Can we increase the interval to like 100ms or something? Same for all other `Eventually` uses in this PR. Ideally no test even takes near a second (though I know it happens sometimes)

Reply:
This is great, thanks!